### PR TITLE
Highlight if log comes from custom component

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Improve logging panel:
 - fall back to reading source path to extract integration if generic logger is used
 - show in overview if it's a custom component
 - show in detailed view if it's a custom component
 - don't show a link to the documentation if the custom component links to the HA website as that is never correct, we don't ship custom components.

image

Screenshot 2021-04-13 at 17 47 07

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/logs/dialog-system-log-detail.ts [hunk 1]
```diff
@@ -88,6 +97,13 @@ class DialogSystemLogDetail extends LitElement {
         </ha-header-bar>
         <div class="contents">
           <p>
+            ${isCustomComponentError(item)
+              ? html`<p>
+                  ${this.hass.localize(
+                    "ui.panel.config.logs.error_from_custom_component"
```
Comment (ludeeus) [inline src/panels/config/logs/dialog-system-log-detail.ts:103]:
```suggestion
                    "ui.panel.config.logs.error_from_custom_integration"
```

File: src/translations/en.json [hunk 2]
```diff
@@ -1067,7 +1067,9 @@
             "warning": "WARNING",
             "info": "INFO",
             "debug": "DEBUG"
-          }
+          },
+          "custom_component": "custom component",
+          "error_from_custom_component": "This error originated from a custom component."
```
Comment (ludeeus) [inline src/translations/en.json:1072]:
```suggestion
          "error_from_custom_integration": "This error originated from a custom integration."
```

File: src/panels/config/logs/system-log-card.ts [hunk 3]
```diff
@@ -78,10 +79,16 @@ export class SystemLogCard extends LitElement {
...
                               ${integrations[idx]
-                                ? domainToName(
+                                ? `${domainToName(
                                     this.hass!.localize,
                                     integrations[idx]!
-                                  )
+                                  )}${
+                                    isCustomComponentError(item)
+                                      ? ` (${this.hass.localize(
+                                          "ui.panel.config.logs.custom_component"
```
Comment (ludeeus) [inline src/panels/config/logs/system-log-card.ts:88]:
```suggestion
                                          "ui.panel.config.logs.custom_integration"
```

File: src/translations/en.json [hunk 4]
```diff
@@ -1067,7 +1067,9 @@
             "warning": "WARNING",
             "info": "INFO",
             "debug": "DEBUG"
-          }
+          },
+          "custom_component": "custom component",
```
Comment (ludeeus) [inline src/translations/en.json:1071]:
```suggestion
          "custom_integration": "custom integration",
```

**Review by ludeeus — APPROVED** [review]

**Comment by spacegaier** [comment]
To make the line in the log details popup stand out more ("... originated from custom integration..."), we could in front of it add the same icon that will be used in PR https://github.com/home-assistant/frontend/pull/8911 as the custom integration indicator.

File: src/translations/en.json (hunk 2 above)
Comment (frenck) [reply src/translations/en.json:1072]: I think we should add a bit on what it means? E.g., if-this-is-an-issue-report-it-with-the-author-of-this-custom-integration-ish.

**Comment by balloob** [comment]
I don't think that it needs to stand-out more on the overview page, it's just something that needs to standout when you open the details. So I will add the icon to the overview page.

**Comment by balloob** [comment]
image

**Pull request merged**